lib/puppet/defaults.rb

```diff
@@ -1286,7 +1286,20 @@ def self.valid_file_checksum_types
       :desc       => "File that provides mapping between custom SSL oids and user-friendly names"
     },
     :basemodulepath => {
-      :default => "$codedir/modules#{File::PATH_SEPARATOR}/opt/puppetlabs/puppet/modules",
+      :default => lambda do
+        installdir =
+          if Puppet::Util::Platform.windows?
+            Facter.value(:env_windows_installdir) || 'C:/Program Files/Puppet Labs/Puppet'
```

Comment:
This will work if a user decides to install the MSI under a directory that isn't `C:/Program Files`?